pkg/kubelet/util/csr/csr.go

```diff
@@ -64,24 +64,21 @@ func RequestNodeCertificate(client certificatesclient.CertificateSigningRequestI
 	})
 	if err != nil {
 		return nil, fmt.Errorf("cannot create certificate signing request: %v", err)
-
 	}
 
 	// Make a default timeout = 3600s.
 	var defaultTimeoutSeconds int64 = 3600
-	resultCh, err := client.Watch(metav1.ListOptions{
+	certWatch, err := client.Watch(metav1.ListOptions{
 		Watch:          true,
 		TimeoutSeconds: &defaultTimeoutSeconds,
 		FieldSelector:  fields.OneTermEqualSelector("metadata.name", req.Name).String(),
 	})
 	if err != nil {
 		return nil, fmt.Errorf("cannot watch on the certificate signing request: %v", err)
 	}
+	defer certWatch.Stop()
 
-	var status certificates.CertificateSigningRequestStatus
-	ch := resultCh.ResultChan()
-
-	for {
+	for ch := certWatch.ResultChan(); ; {
```

Comment:
ah, missed the `; ;`. that's not a common pattern (there's only one other instance in kubernetes), and it doesn't buy us much since the function scope ends one line after the loop.